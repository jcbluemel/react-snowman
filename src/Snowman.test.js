import { render, fireEvent } from "@testing-library/react";
import Snowman from "./Snowman";
import img0 from "./0.png";
import img1 from "./1.png";
import img2 from "./2.png";


it("ends game with loss if max guesses reached", function () {
  const { container } = render(
    <Snowman
      images={[img0, img1, img2]}
      words={["test"]}
      maxWrong={2}
    />
  )

  const aBtn = container.querySelector('button[value="a"]');
  const bBtn = container.querySelector('button[value="b"]');

  fireEvent.click(aBtn);
  fireEvent.click(bBtn);

  expect(aBtn).not.toBeInTheDocument();
  expect(container).toContainHTML("You lose");
})