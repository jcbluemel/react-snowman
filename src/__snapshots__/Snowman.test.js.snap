// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ends game with loss if max guesses reached 1`] = `
<div>
  <div
    class="Snowman"
  >
    <img
      alt="2"
      src="2.png"
    />
    <p>
      Number wrong: 
      2
    </p>
    <p
      class="Snowman-word"
    >
      _
      _
      _
      _
    </p>
    <p>
      You lose: test
    </p>
    <button>
      Restart
    </button>
  </div>
</div>
`;
